accounts/activate.php

```diff
@@ -71,6 +71,8 @@
 }
 
 // Verify we can create the user's forum account, and bail if we can't.
+/** @var NonactivatedUser */
+$user = $user;
```

Comment:
What's going on here?

Reply:
At the time it was the only way I could think of telling PHPStan that `$user` is in fact a `NonactivatedUser`.

Now, I realize that it's probably an incorrect PHPDoc annotation on `load_from_id`. That function has the docs 
```php
/**
  * Load a NonactivatedUser record by ID
  * ...
  * @return object NonactivatedUser
  */
  public static function load_from_id($id) { ... }
```

the `@return object NonactivatedUser` should be `@return NonactivatedUser`

Reply:
Actually, no, I'm wrong. I'd fixed up the `@return NonactivatedUser` already. The actual problem is that PHPStan interprets the fact that the only initialization of `$user` occurs in a fallible `try` block, so it interprets the type as being as generic as possible.

```
  try {
      $user = NonactivatedUser::load_from_id($reg_token);
  } catch (NonexistentNonactivatedUserException $exception) { ... }
 ```

The workaround is to explicitly assign `$user = null;` in the `catch`. Which, logically means the type of `$user` should be `NonactiveUser|null` after the try/catch block, but PHPStan silently ignores the `|null` part in this context.